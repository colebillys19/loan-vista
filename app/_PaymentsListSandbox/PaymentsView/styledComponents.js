import styled from 'styled-components';

import { StyledTableHead } from 'components/_base-ui/GradientListTable';

export const BaseTableData = styled.td`
  padding: 0;
  padding-bottom: ${({ isExpanded }) => (isExpanded ? '16.6rem' : '0')};
  position: relative;
`;

/* eslint-disable prettier/prettier */
export const CustomTableHead = styled(StyledTableHead)`
  & th:nth-of-type(1) {}
  & th:nth-of-type(2) {}
  & th:nth-of-type(3) {}
  & th:nth-of-type(4) {}
  & th:nth-of-type(5) {}
  & th:nth-of-type(6) {}
  & th:nth-of-type(7) {
    width: 2.4rem;
  }
`;
