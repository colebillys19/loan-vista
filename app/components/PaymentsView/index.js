/**
 * PaymentsView
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import ListFilter from 'components/ListFilter';
import PaymentsList from 'components/PaymentsList';
import ConditionalRender from 'components/_base-ui/ConditionalRender';
import ListRenderFallback from 'components/_base-ui/ListRenderFallback';
import TableHeadBorder from 'components/_base-ui/TableHeadBorder';
import TabWrapper from 'components/_base-ui/TabWrapper';

const PaymentsView = ({
  dispatchFetchPaymentsData,
  error,
  fetchParams,
  loading,
  paymentsData,
  sortValues,
}) => {
  const noListData = !loading && paymentsData.length === 0;

  return (
    <TabWrapper aria-labelledby="payments-tab" id="payments-view">
      <ListFilter
        fetchParams={fetchParams}
        dispatchFetchData={dispatchFetchPaymentsData}
      />
      <TableHeadBorder hideBottom={loading || noListData} />
      <ConditionalRender
        Component={
          <PaymentsList
            paymentsData={paymentsData}
            dispatchFetchPaymentsData={dispatchFetchPaymentsData}
            loading={loading}
            sortValues={sortValues}
          />
        }
        FallbackComponent={<ListRenderFallback error={error} />}
        shouldRender={!error && !noListData}
      />
    </TabWrapper>
  );
};

PaymentsView.propTypes = {
  dispatchFetchPaymentsData: T.func.isRequired,
  error: T.oneOfType([T.bool, T.string]).isRequired,
  fetchParams: T.object.isRequired,
  loading: T.bool.isRequired,
  paymentsData: T.array.isRequired,
  sortValues: T.object.isRequired,
};

export default PaymentsView;
