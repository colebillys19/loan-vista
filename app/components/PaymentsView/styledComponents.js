import styled from 'styled-components';

import { StyledTableHead } from 'components/_shared/GradientListTable';

export const BaseTableData = styled.td`
  padding: 0;
  padding-bottom: ${({ isExpanded }) => (isExpanded ? '16.6rem' : '0')};
  position: relative;
`;

export const PaymentsTableHead = styled(StyledTableHead)`
  & th:nth-of-type(1) {
    width: 13.2rem;
  }
  & th:nth-of-type(3) {
    width: 11.6rem;
  }
  & th:nth-of-type(4) {
    width: 13.6rem;
  }
  & th:nth-of-type(5) {
    width: 13.3rem;
  }
  & th:nth-of-type(6) {
    width: 9.4rem;
  }
  & th:nth-of-type(7) {
    width: 2.4rem;
  }
`;
