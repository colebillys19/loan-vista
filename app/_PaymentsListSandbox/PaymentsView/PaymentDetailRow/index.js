import React from 'react';
import T from 'prop-types';

import { StyledTableRow } from 'components/_base-ui/GradientListTable';

import { BaseTableData } from '../styledComponents';
import { CELL_ARR } from './constants';
import {
  Anchor,
  ContentContainer,
  StyledTwoColumnList,
} from './styledComponents';

/* eslint-disable react/no-array-index-key */
const PaymentDetailRow = ({ data }) => (
  <StyledTableRow>
    <BaseTableData>
      <Anchor>
        <ContentContainer>
          {data.map((listData, i) => (
            <StyledTwoColumnList
              key={i}
              data={listData}
              numRows={6}
              renderLoading={false}
            />
          ))}
        </ContentContainer>
      </Anchor>
    </BaseTableData>
    {CELL_ARR.map((v, i) => (
      <BaseTableData key={i} />
    ))}
  </StyledTableRow>
);

PaymentDetailRow.propTypes = {
  data: T.arrayOf(T.shape({ listData: T.array })).isRequired,
};

export default PaymentDetailRow;
