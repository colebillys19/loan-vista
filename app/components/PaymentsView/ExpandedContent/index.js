import React from 'react';
import T from 'prop-types';

import { Container, StyledTwoColumnList } from './styledComponents';

/* eslint-disable react/no-array-index-key */
const PaymentDetailRow = ({ data }) => (
  <Container>
    {data.map((listData, i) => (
      <StyledTwoColumnList key={i} data={listData} numRows={6} />
    ))}
  </Container>
);

PaymentDetailRow.propTypes = {
  data: T.arrayOf(T.shape({ listData: T.array })).isRequired,
};

export default PaymentDetailRow;
