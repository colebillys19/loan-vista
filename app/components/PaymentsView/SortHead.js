/**
 * TableHead
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import ListSortButton from 'components/_shared/ListSortButton';
import {
  StyledTableHeader,
  StyledTableRow,
} from 'components/_shared/GradientListTable';

import { PaymentsTableHead } from './styledComponents';

const SortHead = ({
  dispatchFetchPaymentsData,
  headers,
  lastSortCol,
  lastSortOrder,
  sortLoading,
}) => {
  const handleSortClick = (header) => {
    const newSortCol = header;
    const newSortOrder =
      header === lastSortCol && lastSortOrder === 'desc' ? 'asc' : 'desc';

    dispatchFetchPaymentsData(newSortCol, newSortOrder);
  };

  return (
    <PaymentsTableHead>
      <StyledTableRow>
        {headers.map((header) => (
          <StyledTableHeader key={header} scope="col">
            <ListSortButton
              isActive={header === lastSortCol}
              isAscending={header === lastSortCol && lastSortOrder === 'asc'}
              loading={sortLoading === header}
              onClick={() => handleSortClick(header)}
              text={header}
            />
          </StyledTableHeader>
        ))}
        <StyledTableHeader scope="col" />
      </StyledTableRow>
    </PaymentsTableHead>
  );
};

SortHead.propTypes = {
  dispatchFetchPaymentsData: T.func.isRequired,
  headers: T.arrayOf(T.string).isRequired,
  lastSortCol: T.string.isRequired,
  lastSortOrder: T.string.isRequired,
  sortLoading: T.oneOfType([T.bool, T.string]).isRequired,
};

export default SortHead;
