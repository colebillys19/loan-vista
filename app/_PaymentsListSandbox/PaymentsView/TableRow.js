import React, { Fragment, useState } from 'react';
import T from 'prop-types';

import {
  StyledTableData,
  StyledTableRow,
} from 'components/_base-ui/GradientListTable';
import ConditionalRender from 'components/_base-ui/ConditionalRender';

import ArrowButton from '../ArrowButton';
import PaymentDetailRow from './PaymentDetailRow';
import { BaseTableData } from './styledComponents';

const TableRow = ({ data: { expandedData, mainData }, headers }) => {
  const [isExpanded, setIsExpanded] = useState(false);

  return (
    <Fragment>
      <StyledTableRow>
        {headers.map((header) => (
          <StyledTableData key={header}>{mainData[header]}</StyledTableData>
        ))}
        <BaseTableData>
          <ArrowButton
            handleClick={() => setIsExpanded(!isExpanded)}
            renderUp={isExpanded}
          />
        </BaseTableData>
      </StyledTableRow>
      <ConditionalRender
        Component={PaymentDetailRow}
        propsToPassDown={{ data: expandedData }}
        shouldRender={isExpanded}
      />
    </Fragment>
  );
};

TableRow.propTypes = {
  data: T.shape({
    date: T.string,
    desc: T.string,
    escrow: T.string,
    interest: T.string,
    principal: T.string,
    total: T.string,
  }),
  headers: T.arrayOf(T.string).isRequired,
};

export default TableRow;
