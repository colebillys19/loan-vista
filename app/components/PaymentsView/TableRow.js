import React, { useState } from 'react';
import T from 'prop-types';

import ArrowButton from 'components/_base-ui/ArrowButton';
import {
  StyledTableData,
  StyledTableRow,
} from 'components/_base-ui/GradientListTable';
import ConditionalRender from 'components/_base-ui/ConditionalRender';

import ExpandedContent from './ExpandedContent';
import { BaseTableData } from './styledComponents';

const TableRow = ({ data: { expandedData, mainData }, headers }) => {
  const [isExpanded, setIsExpanded] = useState(false);

  return (
    <StyledTableRow>
      {headers.map((header) => (
        <StyledTableData key={header}>{mainData[header]}</StyledTableData>
      ))}
      <BaseTableData isExpanded={isExpanded}>
        <ArrowButton
          handleClick={() => setIsExpanded(!isExpanded)}
          renderUp={isExpanded}
        />
        <ConditionalRender
          Component={ExpandedContent}
          propsToPassDown={{ data: expandedData }}
          shouldRender={isExpanded}
        />
      </BaseTableData>
    </StyledTableRow>
  );
};

TableRow.propTypes = {
  data: T.shape({
    date: T.string,
    desc: T.string,
    escrow: T.string,
    interest: T.string,
    principal: T.string,
    total: T.string,
  }),
  headers: T.arrayOf(T.string).isRequired,
};

export default TableRow;
