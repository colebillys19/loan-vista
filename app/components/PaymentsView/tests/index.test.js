import React from 'react';
import { render } from 'react-testing-library';

import PaymentsView from '../index';

describe('<PaymentsView />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<PaymentsView />);
    expect(spy).not.toHaveBeenCalled();
  });

  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<PaymentsView />);
    expect(firstChild).toMatchSnapshot();
  });
});
