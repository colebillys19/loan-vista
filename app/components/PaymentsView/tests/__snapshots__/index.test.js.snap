// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentsView /> Should render and match the snapshot 1`] = `
<div
  class="styledComponents__TableContainer-fhggrt-7 gRfYPI"
>
  <div
    class="GradientBorder-zuziiu-0 ldijcC"
  />
  <div
    class="GradientBorder-zuziiu-0 crasIP"
  />
  <div
    class="GradientBorder-zuziiu-0 kSpIlI"
  />
  <div
    class="GradientBorder-zuziiu-0 cgqNXZ"
  />
  <table
    class="styledComponents__StyledTable-fhggrt-1 fYoUQb"
  >
    <thead
      class="styledComponents__StyledTableHead-fhggrt-4 styledComponents__PaymentsTableHead-sc-1kvx6rk-1 eiiNCi"
    >
      <tr
        class="styledComponents__StyledTableRow-fhggrt-6 fSdBuc"
      >
        <th
          class="styledComponents__StyledTableHeader-fhggrt-5 bDgOie"
          scope="col"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1723w1c-0 guxOHO MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              date
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="height: 0.8rem; width: 0.8rem;"
                  viewBox="7 7 10 10"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M7 10l5 5 5-5z"
                    fill="rgb(136, 136, 136)"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          class="styledComponents__StyledTableHeader-fhggrt-5 bDgOie"
          scope="col"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1723w1c-0 guxOHO MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              desc
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="height: 0.8rem; width: 0.8rem;"
                  viewBox="7 7 10 10"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M7 10l5 5 5-5z"
                    fill="rgb(136, 136, 136)"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          class="styledComponents__StyledTableHeader-fhggrt-5 bDgOie"
          scope="col"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1723w1c-0 guxOHO MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              total
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="height: 0.8rem; width: 0.8rem;"
                  viewBox="7 7 10 10"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M7 10l5 5 5-5z"
                    fill="rgb(136, 136, 136)"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          class="styledComponents__StyledTableHeader-fhggrt-5 bDgOie"
          scope="col"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1723w1c-0 guxOHO MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              principal
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="height: 0.8rem; width: 0.8rem;"
                  viewBox="7 7 10 10"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M7 10l5 5 5-5z"
                    fill="rgb(136, 136, 136)"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          class="styledComponents__StyledTableHeader-fhggrt-5 bDgOie"
          scope="col"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1723w1c-0 guxOHO MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              interest
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="height: 0.8rem; width: 0.8rem;"
                  viewBox="7 7 10 10"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M7 10l5 5 5-5z"
                    fill="rgb(136, 136, 136)"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          class="styledComponents__StyledTableHeader-fhggrt-5 bDgOie"
          scope="col"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1723w1c-0 guxOHO MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              escrow
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="height: 0.8rem; width: 0.8rem;"
                  viewBox="7 7 10 10"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M7 10l5 5 5-5z"
                    fill="rgb(136, 136, 136)"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          class="styledComponents__StyledTableHeader-fhggrt-5 bDgOie"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      class="styledComponents__StyledTableBody-fhggrt-2 faBsMU"
    >
      <tr
        class="styledComponents__StyledTableRow-fhggrt-6 fSdBuc"
      >
        <td
          class="styledComponents__StyledTableData-fhggrt-3 ksTgJN"
        >
          date
        </td>
        <td
          class="styledComponents__StyledTableData-fhggrt-3 ksTgJN"
        >
          desc
        </td>
        <td
          class="styledComponents__StyledTableData-fhggrt-3 ksTgJN"
        >
          total
        </td>
        <td
          class="styledComponents__StyledTableData-fhggrt-3 ksTgJN"
        >
          principal
        </td>
        <td
          class="styledComponents__StyledTableData-fhggrt-3 ksTgJN"
        >
          interest
        </td>
        <td
          class="styledComponents__StyledTableData-fhggrt-3 ksTgJN"
        >
          escrow
        </td>
        <td
          class="styledComponents__BaseTableData-sc-1kvx6rk-0 dSjhVg"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root styledComponents__StyledIconButton-sc-1g4y868-0 ChMWt"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="height: 1.2rem; width: 1.2rem;"
                viewBox="7 7 10 10"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 10l5 5 5-5z"
                  fill="rgb(80, 103, 55)"
                />
              </svg>
            </span>
          </button>
        </td>
      </tr>
    </tbody>
    <thead>
      <tr>
        <td
          class="styledComponents__TableFooterCell-fhggrt-8 gtQmvA"
        >
          hidden text
        </td>
      </tr>
    </thead>
  </table>
</div>
`;
